modules/writing_observer/writing_observer/awe_nlp.py

```diff
@@ -202,52 +204,104 @@ async def process_texts_parallel(texts, options=None):
 
 async def process_texts(writing_data, options=None, mode=RUN_MODES.MULTIPROCESSING):
     '''
-    Process texts with caching
-    1. Create hash of text
-    2. Fetch cache data
-    3. Check to see if all options are present
-        - Yes? Add to results
-        - No? Record missing options `needed_options` and add to `need_processing`
-    4. If we need to process anything process it
-    5. Update results/cache
+    Caching:
```

Comment:
I would like to see a diagram of the workflow of this code. The diagram should be included in this docstring so new developers are easily able to understand **exactly** how the cache is expected to work.